File: app/src/main/java/com/android/lowes/activities/WeatherDetailActivity.java
package com.android.lowes.activities;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.android.lowes.R;
import com.android.lowes.models.WeatherDetails;

public class WeatherDetailActivity extends AppCompatActivity {


    // UI components

    private int position;
    private TextView temp, feelsLike,main, description ;

    private static final String TAG = "WeatherDetailActivity";

    private String temperature, feelsTemperature;
    private int tem,feelTem;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_weather_detail);
        temp = findViewById(R.id.temp);
        feelsLike = findViewById(R.id.feels_like);
        main = findViewById(R.id.main);
        description = findViewById(R.id.description);
        Bundle bundle = getIntent().getExtras();
        temp.setText(bundle.getString("temp"));
        feelsLike.setText("feels like : "+bundle.getString("feelsLike"));
        main.setText(bundle.getString("main"));
        description.setText(bundle.getString("description"));
    }


//    private void getIncomingIntent(){
////        if(getIntent().hasExtra("WeatherDetails") && getIntent().hasExtra("position")){
////            WeatherDetails weather ;
////            Bundle bundle = getIntent().getExtras();
////            position = bundle.getInt("position");
////
////            weather = getIntent().getParcelableExtra("WeatherDetails");
////            Log.d(TAG, "getIncomingIntent: " + weather.getWeather().get(position));
////        }
//    }
}

File: app/src/main/java/com/android/lowes/activities/WeatherDetailActivity.kt
package com.android.lowes.activities

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.android.lowes.R

class WeatherDetailActivity : AppCompatActivity() {
    // UI components
    private val position = 0
    private var temp: TextView? = null
    private var feelsLike: TextView? = null
    private var main: TextView? = null
    private var description: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_weather_detail)
        temp = findViewById(R.id.temp)
        feelsLike = findViewById(R.id.feels_like)
        main = findViewById(R.id.main)
        description = findViewById(R.id.description)
        val bundle = intent.extras
        temp?.text = (bundle!!.getString("temp"))
        feelsLike?.text = ("feels like : " + bundle.getString("feelsLike"))
        main?.text = (bundle.getString("main"))
        description?.text = (bundle.getString("description"))
    }

    companion object {
        private const val TAG = "WeatherDetailActivity"
    }
}